ee/app/livechat-enterprise/server/api/lib/units.js

```diff
@@ -1,12 +1,17 @@
+import s from 'underscore.string';
+
 import { hasPermissionAsync } from '../../../../../../app/authorization/server/functions/hasPermission';
 import LivechatUnit from '../../../../models/server/models/LivechatUnit';
 import LivechatUnitMonitors from '../../../../models/server/models/LivechatUnitMonitors';
 
-export async function findUnits({ userId, pagination: { offset, count, sort } }) {
+export async function findUnits({ userId, text, pagination: { offset, count, sort } }) {
 	if (!await hasPermissionAsync(userId, 'manage-livechat-units')) {
 		throw new Error('error-not-authorized');
 	}
-	const cursor = LivechatUnit.find({}, {
+
+	const query = { ...text && { $or: [{ name: new RegExp(s.escapeRegExp(text), 'i') }, { _id: new RegExp(s.escapeRegExp(text), 'i') }] } };
```

Comment:
Why are you finding units by _id using the text parameter?
IMO filtering  `name` field only is enough.